c10/cuda/CUDAStream.cpp

```diff
@@ -43,7 +53,7 @@ static constexpr unsigned int kDefaultFlags = cudaStreamNonBlocking;
 
 // Default streams
 static std::once_flag init_flag;
-static std::vector<CUDAStreamInternals> default_streams;
+static std::unique_ptr<CUDAStreamInternals[]> default_streams;
```

Comment:
If this is a static global variable, unique_ptr is probably not necessary especially if we know the compile time max default streams one will use (like, 16 as max number of gpus?)
Also, note that std::unique_ptr is not trivially destructible and thus is discouraged to be used in global variables (c.f. https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables)